### PR TITLE
0.8.0: Fix set insert returned value

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
Bumping to 0.8.0, because `Set` API has changed to reflect the comment and be similar to `HashSet` API.
Small refactoring, to remove unnecessary storage writes.

Fixes:
#126 

## Test plan
- Updated tests to match the new logic

### CODE REVIEW

File: near-sdk/src/collections/set.rs [hunk 1]
```diff
@@ -63,27 +63,21 @@ impl<T> Set<T> {
         env::storage_has_key(&index_lookup)
     }
 
-    /// Adds a value to the set. If the set did not have this value present, `true` is returned. If
-    /// the set did have this value present, `false` is returned. Note, the elements that have the same hash value are undistinguished by
-    /// the implementation.
+    /// Adds a value to the set.
+    /// If the set did not have this value present, `true` is returned.
+    /// If the set did have this value present, `false` is returned.
+    /// Note, the elements that have the same hash value are undistinguished by the implementation.
```
Comment (nearmax) [inline near-sdk/src/collections/set.rs:69]: Since we do not hash state keys this comment is incorrect.

File: near-sdk/src/collections/vector.rs [hunk 2]
```diff
@@ -73,18 +73,7 @@ impl<T> Vector<T> {
             }
         } else {
             let lookup_key = self.index_to_lookup_key(index);
-            let last_lookup_key = self.index_to_lookup_key(self.len - 1);
-            self.len -= 1;
```
Comment (nearmax) [inline near-sdk/src/collections/vector.rs:77]: This breaks the `Vector::len()` because we need to store the real decreased length in `self.len`

Comment (evgenykuzyakov) [reply near-sdk/src/collections/vector.rs:77]: `pop_raw` decreases length

File: near-sdk/src/collections/set.rs (hunk 1 above)
Comment (evgenykuzyakov) [reply near-sdk/src/collections/set.rs:69]: removed

**Comment by nearmax** [comment]
Please publish the libs.

**Pull request merged**